### PR TITLE
feat: top files collection

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

**Review by Hardikl — DISMISSED** [review]

File: conf/restperf/9.12.0/volume.yaml [hunk 1]
```diff
@@ -50,11 +50,14 @@ plugins:
 
 #  - VolumeTopClients:
 #      # The maximum number of volumes to consider for top client metrics.
-#      # The actual maximum value is capped at 50, even if a higher number is specified.
+#      # The actual maximum value for max_volumes is capped at 50, even if a higher number is specified.
```
Comment (cgrinds) [inline conf/restperf/9.12.0/volume.yaml:53]: This value is capped at 50, even if a higher number is specified.

File: conf/restperf/9.12.0/volume.yaml [hunk 2]
```diff
@@ -50,11 +50,14 @@ plugins:
...
 #      # When enabled, this plugin will collect read/write operations and throughput metrics for the top clients of each volume.
 #      - schedule:
 #          - data: 1h  # This value should be a multiple of the poll duration. By default, Harvest will check once an hour to see how many volumes have activity_tracking.state set to on.
 #      - max_volumes: 5
+#      - objects:
+#          - client  # remove to disable the collection of read/write operations and throughput metrics for the top clients.
```
Comment (cgrinds) [inline conf/restperf/9.12.0/volume.yaml:59]: # collect read/write operations and throughput metrics for the top clients

File: conf/restperf/9.12.0/volume.yaml [hunk 3]
```diff
@@ -50,11 +50,14 @@ plugins:
...
+#          - file    # remove to disable the collection of read/write operations and throughput metrics for the top files.
```
Comment (cgrinds) [inline conf/restperf/9.12.0/volume.yaml:60]: # collect read/write operations and throughput metrics for the top files

File: pkg/util/util.go [hunk 4]
```diff
@@ -471,3 +471,12 @@ func SafeConvertToInt32(in int) (int32, error) {
 	}
 	return int32(in), nil // #nosec G115
 }
+
+func Contains(slice []string, item string) bool {
```
Comment (cgrinds) [inline pkg/util/util.go:475]: not needed, can use `slices.Contains` instead

File: cmd/collectors/restperf/plugins/volumetopmetrics/volumetopmetrics.go [hunk 5]
```diff
@@ -125,12 +148,31 @@ func (t *TopClients) Init() error {
...
 		}
 	}
+
+	// enable client and file metrics collection by default
+	t.clientMetricsEnabled = true
+	t.fileMetricsEnabled = true
+
+	if objects := t.Params.GetChildS("objects"); objects != nil {
+		o := objects.GetAllChildContentS()
+		if !util.Contains(o, "client") {
```
Comment (cgrinds) [inline cmd/collectors/restperf/plugins/volumetopmetrics/volumetopmetrics.go:158]: slices.Contains(o, "client")

File: cmd/collectors/restperf/plugins/volumetopmetrics/volumetopmetrics.go [hunk 6]
```diff
@@ -125,12 +148,31 @@ func (t *TopClients) Init() error {
...
+			t.clientMetricsEnabled = false
+		}
+		if !util.Contains(o, "file") {
```
Comment (cgrinds) [inline cmd/collectors/restperf/plugins/volumetopmetrics/volumetopmetrics.go:161]: slices.Contains(o, "file")

**Review by Hardikl — APPROVED** [review]

**Review by cgrinds — APPROVED** [review]

**Pull request merged**